src/rdiff_backup/Main.py

```diff
@@ -416,6 +440,13 @@ def _Main(arglist):
     cmdpairs = SetConnections.get_cmd_pairs(_args, _remote_schema, _remote_cmd)
     Security.initialize(_action or "mirror", cmdpairs)
     rps = list(map(SetConnections.cmdpair2rp, cmdpairs))
+
+    # if any of the remote paths is None, we have an error.
+    # We continue to test-server so that all connections can be tested at once.
+    if any(map(lambda x: x is None, rps)) and _action != "test-server":
```

Comment:
How is this related to the API version ?

Reply:
It's related to the "None" discussion after: basically, by sending None, I can test _all_ servers and don't need to stop each time one fails, forcing the user to call again `rdiff-backup --test-server` to make sure they identified all failing servers. I had to touch this part of the code because I added the handling of the API version between client and server.